stix2/sources/__init__.py

```diff
@@ -28,14 +24,100 @@ def make_id():
     return str(uuid.uuid4())
 
 
-class DataStore(object):
+def apply_common_filters(stix_objs, query):
+    """Evaluate filters against a set of STIX 2.0 objects.
+
+    Supports only STIX 2.0 common property fields
+
+    Args:
+        stix_objs (list): list of STIX objects to apply the query to
+
+        query (set): set of filters (combined form complete query)
+
+    Returns:
+        (list): list of STIX objects that successfully evaluate against
+            the query.
+
     """
+    filtered_stix_objs = []
+
+    # evaluate objects against filter
+    for stix_obj in stix_objs:
+        clean = True
+        for filter_ in query:
+            # skip filter as filter was identified (when added) as
+            # not a common filter
+            if filter_.field not in STIX_COMMON_FIELDS:
+                raise ValueError("Error, field: {0} is not supported for filtering on.".format(filter_.field))
+
+            # For properties like granular_markings and external_references
+            # need to break the first property from the string.
+            if "." in filter_.field:
+                field = filter_.field.split(".")[0]
+            else:
+                field = filter_.field
+
+            # check filter "field" is in STIX object - if cant be
+            # applied due to STIX object, STIX object is discarded
+            # (i.e. did not make it through the filter)
+            if field not in stix_obj.keys():
+                clean = False
+                break
+
+            match = STIX_COMMON_FILTERS_MAP[filter_.field.split('.')[0]](filter_, stix_obj)
+            if not match:
+                clean = False
+                break
+            elif match == -1:
+                raise ValueError("Error, filter operator: {0} not supported for specified field: {1}".format(filter_.op, filter_.field))
+
+        # if object unmarked after all filters, add it
+        if clean:
+            filtered_stix_objs.append(stix_obj)
+
+    return filtered_stix_objs
+
+
+def deduplicate(stix_obj_list):
+    """Deduplicate a list of STIX objects to a unique set
```

Comment:
Does this function make sense outside of sources? Maybe just put in the `stix2.utils` module?

Reply:
Will relocate